charts/shoot-core/components/charts/kube-proxy/templates/kube-proxy-init-script.yaml

```
@@ -0,0 +1,25 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-proxy-init-script
+  namespace: kube-system
+data:
+  init.sh: |
+    #!/bin/sh
+    OLD_KUBE_PROXY_MODE=$(cat /kube-proxy-mode);
+    if [ -z ${OLD_KUBE_PROXY_MODE} ] || [ ${OLD_KUBE_PROXY_MODE} = ${KUBE_PROXY_MODE} ]; then
+      echo ${KUBE_PROXY_MODE} >/kube-proxy-mode;
+      exit 0
+    fi;
+    {{- if semverCompare "< 1.17" .Values.kubernetesVersion }}
+    /hyperkube
+    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
+    proxy
+    {{- else }}
+    kube-proxy
+    {{- end }}
+    {{- else }}
+    /usr/local/bin/kube-proxy
+    {{- end }} --v=2 --cleanup --proxy-mode=${OLD_KUBE_PROXY_MODE} || exit $?;
+    echo ${KUBE_PROXY_MODE} >/kube-proxy-mode;
```

```suggestion
    OLD_KUBE_PROXY_MODE="$(cat "$1")";
    if [ -z "${OLD_KUBE_PROXY_MODE}" ] || [ "${OLD_KUBE_PROXY_MODE}" == "${KUBE_PROXY_MODE}" ]; then
      echo "${KUBE_PROXY_MODE}" > "$1";
      exit 0
    fi;
    {{- if semverCompare "< 1.17" .Values.kubernetesVersion }}
    /hyperkube
    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
    proxy
    {{- else }}
    kube-proxy
    {{- end }}
    {{- else }}
    /usr/local/bin/kube-proxy
    {{- end }} --v=2 --cleanup --proxy-mode="${OLD_KUBE_PROXY_MODE}" || exit $?;
    echo "${KUBE_PROXY_MODE}" > "$1";
```

About this line:

> `--proxy-mode="${OLD_KUBE_PROXY_MODE}" || exit $?;`

What's the rationale behind it? Why do you need this `|| exit` part?

---

If the clean-up fails, it has to be retried while the file on the host still contains the old kube-proxy mode.

---

Sure, but can't we just set the `-e` flag, i.e., `#!/bin/sh -e`? Then line 25 won't be executed if the cleanup process fails.

---

Sure, with `-e` it is the same behavior.